server/services/user.service.js

```diff
@@ -32,18 +32,56 @@ export const comparePassword = async (password, hashedPassword) => {
   }
 };
 
+export const getUserByReferralCode = async (referralCode, fields = null) =>
+  User.findOne({ referralCode }).select(fields);
+
+export const generateReferralCode = async (firstName) => {
+  const name = firstName.substring(0, 3).toLowerCase();
+  let code = name;
+  let charsToAdd = 4;
```

Comment:
This line and subsequent lines can be fixed into a single line
```suggestion
  const charsToAdd = firstName.length < 3 ? 7 - firstName.length : 4;
```